sympy/solvers/solveset.py

```diff
@@ -1754,6 +1763,31 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _solve_as_lambert(lhs, rhs, symbol, domain):
+    r"""
+    Computes the equations whose solutions will be given in terms
+    of the `LambertW` function. Returns `None` if it fails doing so.
+
+    Example
+    =======
+
+    >>> from sympy.solvers.solveset import _solve_as_lambert
+    >>> from sympy import symbols, exp, S
+    >>> x = symbols('x')
+    >>> _solve_as_lambert(x*exp(x), 1, x, S.Reals)
+    FiniteSet(LambertW(1))
+
+    """
+    poly = lhs.as_poly()
+    gens = _filtered_gens(poly, symbol)
+    try:
+        soln = _solve_lambert(lhs - rhs, symbol, gens)
+    except NotImplementedError:
```

Comment:
If it is not implemented then is not it user should be knowing this rather than returning none ?